File: backend/app/schemas/vessel.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime, date
from enum import Enum


# Match the SQLAlchemy enum
class VesselStatusEnum(str, Enum):
    active = "active"
    inactive = "inactive"
    pending = "pending"


class VesselBase(BaseModel):
    vessel_name: str
    registration_number: str
    weight_capacity: float
    entry_date: date
    status: Optional[VesselStatusEnum] = VesselStatusEnum.pending


class VesselCreate(VesselBase):
    created_by: Optional[int]


class VesselUpdate(BaseModel):
    vessel_name: Optional[str] = None
    registration_number: Optional[str] = None
    weight_capacity: Optional[float] = None
    entry_date: Optional[date] = None
    status: Optional[VesselStatusEnum] = None


class VesselOut(VesselBase):
    id: int
    created_by: Optional[int]
    created_at: Optional[datetime]
    updated_at: Optional[datetime]

    class Config:
        orm_mode = True

File: frontend/src/services/vessel.ts
import axios from './axios';

export interface Vessel {
  id: number;
  vessel_name: string;
  registration_number: string;
  weight_capacity: number;
  entry_date: string; // e.g., "2025-07-25"
  status: 'pending' | 'active' | 'inactive' | string; // Adjust as needed
  created_by: number;
  created_at: string; // ISO timestamp
  updated_at: string; // ISO timestamp
  [key: string]: any; // Optional, for flexibility
}

// Fetch all vessels
export const fetchVesselsApi = async (): Promise<Vessel[]> => {
  const res = await axios.get('/vessels');
  return res.data;
};

// Create new vessel
export const createVesselApi = async (data: Partial<Vessel>) => {
  const res = await axios.post('/vessels', data);
  return res.data;
};

// Update existing vessel
export const updateVesselApi = async (id: number, data: Partial<Vessel>) => {
  const res = await axios.put(`/vessels/${id}`, data);
  return res.data;
};

// Delete vessel
export const deleteVesselApi = async (id: number) => {
  const res = await axios.delete(`/vessels/${id}`);
  return res.data;
};
